test/allocator.cpp

```diff
@@ -1,18 +1,42 @@
 // Copyright 2019-2023 Alfredo A. Correa
+// Copyright 2024 Matt Borland
+// Distributed under the Boost Software License, Version 1.0.
+// https://www.boost.org/LICENSE_1_0.txt
 
-#include <boost/test/unit_test.hpp>
-
-#include <multi/array.hpp>
-#include <multi/pmr.hpp>
+#include <boost/multi/array.hpp>
+#include <boost/multi/pmr.hpp>
 
-#include <multi/detail/static_allocator.hpp>
+#include <boost/multi/detail/static_allocator.hpp>
 
 #include <vector>
 
 #if(not defined(__GLIBCXX__) or (__GLIBCXX__ >= 20210601)) and (not defined(_LIBCPP_VERSION) or (_LIBCPP_VERSION > 14000))
 #include <memory_resource>
 #endif
 
+// Suppress warnings from boost.test
```

Comment:
I am surprised the warning suppression go here. Is this the common way? also they seem to be the same for clang and GNUC, couldn't be compacted into a single `#if defined(__clang__) || defined(__GNUC__)` ?

Reply:
This should be right because I rearranged the include order so that the library is included first and this should just stamp out all the problems from the other boost libraries. Not all versions of clang (e.g. Clang-cl) handle `#pragma GCC` so while it is duplicated it offers diverse support.